tools/Performance/DynamoPerformanceTests/ResultsComparer.cs

```diff
@@ -0,0 +1,376 @@
+            LogResults();
+        }
+
+        /// <summary>
+        /// Compare the two results files and build comparison data
+        /// </summary>
+        private void CreateComparisons()
+        {
+            var baseResults = ImportResultsCSV(BasePath);
+            var newResults = ImportResultsCSV(NewPath);
+
+            Comparisons = new List<BenchmarkComparison>();
+
+            foreach(var result in newResults.Values)
+            {
+                var baseResult = GetBaseResult(result, baseResults);
+                Comparisons.Add(new BenchmarkComparison(baseResult, result));
+            }
+        }
+
+        /// <summary>
+        /// Print comparison results to the console
+        /// </summary>
+        private void LogResults()
+        {
+            var columnWidths = GetColumnWidths();
+
+            // Log header
+            var header = new string[] { "Method", "Graph", "Version", "Mean", "Error", "StdDev" };
+            var separator = new string[header.Length];
+            for (int i = 0; i < header.Length; i++)
+            {
+                header[i] = (header[i] + " ").PadLeft(columnWidths[i], ' ');
+                separator[i] = "".PadLeft(columnWidths[i], '-');
+            }
+            Console.WriteLine("|" + string.Join("|", header) + "|");
+            Console.WriteLine("|" + string.Join("|", separator) + "|");
+            
+            // Log comparisons
+            foreach (var comparison in Comparisons)
+            {
+                Console.ForegroundColor = Console.ForegroundColor == ConsoleColor.Gray ? ConsoleColor.White : ConsoleColor.Gray;
+                comparison.LogComparison(columnWidths);
+            }
+        }
+
+        /// <summary>
+        /// Write the results of this comparison to a csv file
+        /// </summary>
+        /// <param name="filePath"></param>
+        public void WriteResultsToCSV(string filePath)
+        {
+            var rows = GetComparisonDataRows().Select(r => string.Join(",", r));
+            var csv = new StringBuilder();
+            foreach (var row in rows) csv.AppendLine(row);
+            
+            File.WriteAllText(filePath, csv.ToString());
+        }
+        
+        private Dictionary<string, BenchmarkResult> ImportResultsCSV(string csvPath)
+        {
+            // Get csv
+            csvPath = PerformanceTestHelper.GetFullPath(csvPath);
+            TextFieldParser parser = new TextFieldParser(csvPath);
+            parser.HasFieldsEnclosedInQuotes = true;
+            parser.SetDelimiters(",");
+
+            // Get csv header
+            var header = parser.ReadFields();
+
+            // Check that columns we care about exist
+            var columnNames = new string[] { "Method", "Graph", "Mean", "Error", "StdDev" };
+            CheckCSVHeader(header, columnNames, csvPath);
+
+            // Get indices for columns we care about
+            var iMethod = Array.IndexOf(header, columnNames[0]);
```

Comment:
so if the results files change number of columns or order we'll need to remember to change this code or our numbers will be very wrong?

Reply:
No, this will find the column index of “Method” wherever it is in the csv.

Reply:
We only have to change this code if we change the name of a column. For example, “Graph” to “DynamoGraph”.